presto-spark-base/src/main/java/com/facebook/presto/spark/PrestoSparkQueryExecutionFactory.java

```diff
@@ -216,6 +216,7 @@
 {
     private static final Logger log = Logger.get(PrestoSparkQueryExecutionFactory.class);
     public static final String PRESTO_QUERY_ID_CONFIG = "presto_query_id";
+    public static final String SPARK_DYNAMIC_ALLOCATION_MAX_EXECUTORS_CONFIG = "spark.dynamicAllocation.maxExecutors";
```

Comment:
Is this per stage or per query? If it is per query, how do we determine the per-stage one? Or I guess per-stage one doesn't matter because hash partition count can already determine that. Can we add a comment to explain the benefit of this config? (I guess it is to have a lower estimation of the executor count so we don't have to do over allocation?). Also, seems that the spark-related configs will be centralized in `PrestoSparkSettingsRequirements`. @pgupta2 probably have more context on that.

Also, by looking at https://spark.apache.org/docs/latest/configuration.html#dynamic-allocation, do we also need to enable other configs like `spark.dynamicAllocation.enabled` in order to make it useful?

Reply:
@highker : To modify the executor count we need to update sparkContext.conf() and the key for the field is `spark.dynamicAllocation.maxExecutors`. Rather than specifying the key in code I added the SPARK_DYNAMIC_ALLOCATION_MAX_EXECUTORS_CONFIG to reflect the key. This config key is how we update the executor count. Since this functionally is not used in`PrestoSparkSettingsRequirements` we are not adding it there. I will add a note.

Also  by default we enable `spark.dynamicAllocation.enabled` in Meta. That's what I have observed so far.